mcs/class/corlib/System.Runtime.Remoting.Messaging/MonoMethodMessage.cs

```diff
@@ -68,8 +68,38 @@ internal class MonoMethodMessage : IMethodCallMessage, IMethodReturnMessage, IIn
 		internal static String CallContextKey = "__CallContext";
 		internal static String UriKey           = "__Uri";
 
-		[MethodImplAttribute(MethodImplOptions.InternalCall)]
-		internal extern void InitMessage (MonoMethod method, object [] out_args);
+		internal void InitMessage (MonoMethod method, object [] out_args)
+		{
+			this.method = method;
+			ParameterInfo[] paramInfo = method.GetParametersInternal ();
+			int param_count = paramInfo.Length;
```

Comment:
This will call into native code (ves_icall_get_parameter_info), doing lookups in caches, doing locking, so its pretty slow.